### ISSUE TITLE
The rke2 agent does not start

### ISSUE THREAD

**Issue by drylio** [rancher/rke2#3137]
Problem with the rke2 agent. The rke2 server started and everything is fine. But when I try to start the rke2 agent I get info that:

```
 journalctl -u rke2-agent -f

"Waiting to retrieve agent configuration; server is not ready: https://127.0.0.1:6444/v1-rke2/serving-kubelet.crt: 503 Service Unavailable
```

Where is this local IP coming from? Do I need to change it on the agent machine?

**Comment by brandond** [comment]
The agent runs a local load-balancer that routes connections to all available servers; this is why you see a loopback address in the message.

If you're confident that the server address and token are correct on the agent, check the rke2-server logs - are there any errors indicating why the server is not yet ready?

**Comment by drylio** [comment]
Maybe you're right, because on the server side, when I run "journalctl" I find that server not ready: 
"Waiting to retrieve kube-proxy configuration; server is not ready: https://127.0.0.1:9345/v1-rke2/readyz: 500 Internal Server Error"
Why is it taking a local IP and where to change it?

And a question - in the server file "/etc/rancher/rke2/config.yaml"), at the line "server: https://127.0.0.1:6443", should I change this local IP to the server IP? If yes, then stupid question, how :) when I change the IP to the server IP and save the file everything is ok. But if I restart the service "service rke2-server restart" the local IP is returned.
Thanks for the answers, wasted time.

**Comment by brandond** [comment]
> in the server file "/etc/rancher/rke2/config.yaml"), at the line "server: https://127.0.0.1:6443/"
> But if I restart the service "service rke2-server restart" the local IP is returned.

I'm confused - if you have a single server, its configuration file should not have a `server:` entry in it. Are you sure you're not looking at the admin kubeconfig at `/etc/rancher/rke2/rke2.yaml`? The kubeconfig should always point at localhost.

On the agent, the config file (NOT the kubeconfig, agents don't have a kubeconfig) should have a server entry pointing at the server node's LAN IP.

**Comment by xhejtman** [comment]
I am getting this error:
"Waiting to retrieve kube-proxy configuration; server is not ready: https://127.0.0.1:9345/v1-rke2/readyz: 500 Internal Server Error"
as well with 1.24.2-rke2r1. 

It seems it does not start kube-proxy container.

**Comment by xhejtman** [comment]
This is happening if `container-runtime-endpoint: /run/containerd/containerd.sock` external containerd is used. Is this not supported anymore? Or is problem that external containerd version is 1.5, while internal is 1.6?

**Comment by drylio** [comment]
Thank you all for your time and comments,

The biggest problem here seems to be the proxy server. We have tried to install rke2, rancher and longhorn on servers that have direct access to the internet. Everything worked fine. We cloned the same servers into an environment that accesses the internet via a proxy and everything stopped (the rke2 server did not start). The problem is that in the proxy settings it is not enough to describe the standard addresses (github.com, get.rke2.io, etc.) It seems that a lot of different addresses are being addressed, and if they are not described, the rke2-agent, the rke2-server, doesn't start.
